### PR TITLE
Migrate controller to use function SetupWithManager

### PR DESCRIPTION
**Pull request by jottofar**
This was not originally done in the migration to operator-sdk v0.19.3. In that original migration I choose to continue using the legacy leader election method, which is still supported, and was not sure how to do so while also using SetupWithManager. This PR migrates leader election to the controller-runtime’s leader with lease mechanism as well.

Also added back in a couple of kubebuilder:scaffold markers which shouldn't have been deleted

### CODE REVIEW

**Comment by jottofar** [comment]
Not sure why deployment error is poping up now so will retry:
`CreateDeploymentFailed	{"Request.Namespace": "openshift-updateservice", "Request.Name": "example", "error": "Deployment.apps \"example\" is invalid:`

/test operator-e2e

**Comment by jottofar** [comment]
Works locally but took ~7 minutes for operand deployment to be ready so perhaps it was too slow in coming up in Ci env.

File: main.go [hunk 1]
```diff
@@ -73,85 +78,33 @@ func main() {
...
-	uscontroller.AddToManagerFuncs = append(uscontroller.AddToManagerFuncs, func(mgr manager.Manager) error {
-		return uscontroller.Add(mgr, opts)
-	})
-	// Setup UpdateService controllers
-	if err := uscontroller.AddToManager(mgr); err != nil {
-		log.Error(err, "")
+	if err = (&controllers.UpdateServiceReconciler{
+		Client: mgr.GetClient(),
+		Log:    ctrl.Log.WithName("controllers").WithName("UpdateService"),
+		Scheme: mgr.GetScheme(),
```
Comment (wking) [inline main.go:102]: [e2e][1] has lots of logs like:

```
2020-11-12T22:38:21.734Z	ERROR	controller_updateservice	CreateDeploymentFailed	{"Request.Namespace": "openshift-updateservice", "Request.Name": "example", "error": "Deployment.apps \"example\" is invalid: [spec.template.spec.containers[0].image: Required value, spec.template.spec.containers[1].image: Required value]"}
github.com/go-logr/zapr.(*zapLogger).Error
	/go/src/github.com/openshift/cincinnati-operator/vendor/github.com/go-logr/zapr/zapr.go:132
github.com/openshift/cincinnati-operator/controllers.handleErr
	/go/src/github.com/openshift/cincinnati-operator/controllers/updateservice_controller.go:172
github.com/openshift/cincinnati-operator/controllers.(*UpdateServiceReconciler).ensureDeployment
```

To pass in an image you need `OPERAND_IMAGE` handling like [this][2].

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cincinnati-operator/82/pull-ci-openshift-cincinnati-operator-master-operator-e2e/1327005667935916032
[2]: https://github.com/openshift/cincinnati-operator/pull/76/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R85

**Comment by jottofar** [comment]
Have an issue to fix causing test failure

/hold

File: main.go (hunk 1 above)
Comment (wking) [reply main.go]: I've pushed 7122c12 to address this; squash it in if it looks good to you.

**Comment by wking** [comment]
[images][1]:

```
error: build error: failed to pull image: After retrying 2 times, Pull image still failed due to error: while pulling "docker://docker.io/openshift/origin-release:golang-1.13" as "docker.io/openshift/origin-release:golang-1.13"...
```

We should be pulling that from the CI registry...

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cincinnati-operator/82/pull-ci-openshift-cincinnati-operator-master-images/1328406480835055616#1:build-log.txt%3A39

**Comment by LalatenduMohanty** [comment]
Needs a rebase as #86 is merged now.

File: main.go (hunk 1 above)
Comment (jottofar) [reply main.go]: done

**Review by LalatenduMohanty — APPROVED** [review]
/lgtm

**Pull request merged**